### ISSUE TITLE
when I changed the warmup steps from 300 to 30, an error occurred.

### ISSUE THREAD

**Issue by zacario-li** [RangiLyu/nanodet#7]
```
Traceback (most recent call last):
  File "tools/train.py", line 92, in <module>
    main(args)
  File "tools/train.py", line 87, in main
    trainer.run(train_dataloader, val_dataloader, evaluator)
  File "/zzz/prj/nanodet/nanodet/trainer/trainer.py", line 123, in run
    self.warm_up(train_loader)
  File "/zzz/prj/nanodet/nanodet/trainer/trainer.py", line 191, in warm_up
    output, loss, loss_stats = self.run_step(model, batch)
  File "/zzz/prj/nanodet/nanodet/trainer/trainer.py", line 57, in run_step
    loss.backward()
  File "/usr/local/lib/python3.6/dist-packages/torch/tensor.py", line 221, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/usr/local/lib/python3.6/dist-packages/torch/autograd/__init__.py", line 132, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```

**Comment by RangiLyu** [comment]
Can you provide your environment and training settings?

**Comment by wangyin** [comment]
I ran into the similar issue. Solved it by add "requires_grad=True" in the three lines.
https://github.com/RangiLyu/nanodet/blob/5a7a50108d4c6815fc4bbcc382d08c8679143449/nanodet/model/head/gfl_head.py#L306-L308

**Comment by zacario-li** [comment]
> I ran into the similar issue. Solved it by add "requires_grad=True" in the three lines.
> https://github.com/RangiLyu/nanodet/blob/5a7a50108d4c6815fc4bbcc382d08c8679143449/nanodet/model/head/gfl_head.py#L306-L308

@wangyin Love you, bro, it solved the issue.
@RangiLyu Could you figure out why?

**Comment by zacario-li** [comment]
Found some reason which from PyTorch discuss
[tensors are created with requires_grad=False by default](https://discuss.pytorch.org/t/element-0-of-tensors-does-not-require-grad-and-does-not-have-a-grad-fn/32908/28?u=zhijian_li)

**Comment by RangiLyu** [comment]
Thanks. This bug has fixed in commit a444d72
